src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -1473,18 +1509,15 @@ namespace stan {
                    const scope& var_scope,
                    bool& pass, variable_map& vm,
                    std::stringstream& error_msgs) const {
+      pass = true;
       int numdims = expr.expression_type().num_dims();
-      pass = !(vm.exists(name));
-      if (!pass)
-        error_msgs << "ERROR: loop variable already declared."
-                   << " variable name=\"" << name << "\"" << std::endl;
       if (!(numdims > 0)) {
         pass = false;
```

Comment:
Does this produce an error message anywhere? Perhaps that could be informative?

Reply:
I don't see it does, actually. Shouldn't it write to the error stream here, if `pass` is set to `false`?